gen3/bin/report-tool.sh

```diff
@@ -0,0 +1,139 @@
+  eks_version=$(echo -e ${eks_information} | jq -r .cluster.version)
+
+  k8s_version=$(g3kubectl version  -o json)
+
+}
+
+function instances_facts() {
+
+  local instances_name="eks-${vpc_name}"
+  instance_list="$(aws ec2 describe-instances --filter "Name=tag:Name,Values=${instances_name}*")"
+  instance_names="$(echo -e ${instance_list}| jq -r '.Reservations[].Instances[].Tags[] | select( .Key=="Name" ) | .Value')"
+
+  total_instance_count="$(echo -e "${instance_names}" | wc -l)"
+
+  jupyter_instances_count=$(echo -e "${instance_names}" | grep jupyter | wc -l)
+  default_instances_count=$(echo -e "${instance_names}" | grep -v jupyter | wc -l)
+
+  # echo -e ${instance_list} |jq -r '.Reservations[].Instances[] | select( .Tags[].Value=="'${instances_name}-jupyter'") | .Tags[] |select( .Key=="Name" )| .Value' |wc -l
+  jupyter_instances="$(echo -e ${instance_list} |jq -r '.Reservations[].Instances[] | select( .Tags[].Value=="'${instances_name}-jupyter'")')"
+  default_instances="$(echo -e ${instance_list} |jq -r '.Reservations[].Instances[] | select( .Tags[].Value=="'${instances_name}'")')"
+
+  #jupyter_instance_type="$(echo -e ${instance_list} |jq -r '.Reservations[].Instances[] | select( .Tags[].Value=="'${instances_name}-jupyter'") | .InstanceType' |sort |uniq)"
+  jupyter_instance_type="$(echo -e "${jupyter_instances}" |jq .InstanceType | sort | uniq)"
+  #default_instance_type="$(echo -e ${instance_list} |jq -r '.Reservations[].Instances[] | select( .Tags[].Value=="'${instances_name}'") | .InstanceType' |sort |uniq)"
+  default_instance_type="$(echo -e "${default_instances}" |jq .InstanceType | sort | uniq)"
+
+  images_id=$(echo -e ${instance_list} | jq -r .Reservations[].Instances[].ImageId | sort |uniq)
+}
+
+
+function print_report_full() {
+
+  echo
+  echo ${vpc_name}
+  echo -e "EKS version: ${eks_version}"
+  echo -e "K8s version: Client-> $(echo -e ${k8s_version} | jq -r '.clientVersion| "\(.major).\(.minor)"') -- Server-> $(echo -e ${k8s_version} | jq -r '.serverVersion | "\(.major).\(.minor)"')"
+  echo
+  echo -e "Pool\t\tCount\t\tFlavor"
+  echo -e "default\t\t${default_instances_count}\t\t$(echo ${default_instance_type})"
+  echo -e "jupyter\t\t${jupyter_instances_count}\t\t$(echo ${jupyter_instance_type})"
+  echo
+  echo -e "AMIs: "${images_id}
+
+}
+
+## Let's do some admin work to find out the variables to be used here
+BOLD='\e[1;31m'         # Bold Red
+REV='\e[1;32m'       # Bold Green
+
+#Help function
+function HELP {
+  echo -e "${REV}Basic usage:${OFF} ${BOLD}$SCRIPT -r full ${OFF}"\\n
+  echo -e "${REV}The following switches are recognized. $OFF "
+  echo -e "${REV}-r --report ${OFF}  Print a report, valid types are 'full' only ${OFF}."
+#  echo -e "Example: ${BOLD}$SCRIPT -d helloworld -p /opt/py27env/bin -v 2.7 ${OFF}"\\n
+  exit 1
+}
+
+
+function main() {
+
+  case "${1}" in
+    "full")
+      kube_facts
+      instances_facts
+      print_report_full
+      ;;
```

Comment:
not crazy about this pattern of setting global variables for use in other functions.
would be nicer if each function just used local variables, and maybe returned a json blob:
```
kube_facts() {
  ...
  cat - <<EOM
{
    "version": "$version",
   ...
}
EOM
}

...
local k8sInfo=$(kube_facts)
echo "k8s version: $(jq -r .version <<<"$k8sInfo")"
```

we can let it slide this time ... :-p